connectors/citrus-kubernetes/src/main/java/org/citrusframework/kubernetes/config/annotation/KubernetesClientConfigParser.java

```diff
@@ -34,38 +34,58 @@ public class KubernetesClientConfigParser implements AnnotationConfigParser<Kube
     public KubernetesClient parse(KubernetesClientConfig annotation, ReferenceResolver referenceResolver) {
         KubernetesClientBuilder builder = new KubernetesClientBuilder();
 
-        if (StringUtils.hasText(annotation.url())) {
-            builder.url(annotation.url());
-        }
+        if (isClientWithAuthToken(annotation) || isClientWithUsernameAndPassword(annotation) || isClientDefaultConfigured(annotation)) {
```

Comment:
to keep the change smaller and the code more readable, I suggest 
```suggestion
if (!isClientWithAuthToken(annotation) && !isClientWithUsernameAndPassword(annotation) && !isClientDefaultConfigured(annotation)) {
            throw new IllegalArgumentException("Parameters not set correctly - check if either an oauthToke or password and username is set");
}
```

the, continue with the old code.

also, this sanity check does not take invocation from XML into account, as far as I see. it does only handle the annotatoin-case, that's why the parameter is the annotation itself: `KubernetesClientConfig `.